cnf-tests/docgen/cmd/fill.go

```diff
@@ -78,21 +78,49 @@ func fill(xmlFile, descriptionsFile string) {
 
 func fillDescriptions(fileName string, tests TestSuite, currentDescriptions map[string]string) error {
 	missing := false
+	removed := false
+	errMissing := "Found tests with no description"
+	errRemoved := "Found tests that were removed"
+	allTests := make([]string, 0)
 	for _, t := range tests.Testcase {
 		if _, ok := currentDescriptions[t.Name]; !ok {
 			currentDescriptions[t.Name] = emptyPlaceHolder
-			fmt.Printf("The test %s do not have a valid description\n", t.Name)
+			fmt.Printf("The test %s does not have a valid description\n", t.Name)
 			missing = true
 		}
+		allTests = append(allTests, t.Name)
+	}
+	for k := range currentDescriptions {
+		if !sliceContains(allTests, k) {
+			delete(currentDescriptions, k)
+			fmt.Printf("The test %s was removed from the test suite\n", k)
+			removed = true
+		}
 	}
 	jsonData, err := json.MarshalIndent(currentDescriptions, "", "    ")
 	err = ioutil.WriteFile(fileName, jsonData, os.ModePerm)
 	if err != nil {
 		return fmt.Errorf("failed to open the descriptions file %v", err)
 	}
-
+	if missing && removed {
+		return fmt.Errorf("%s, %s", errMissing, errRemoved)
+	}
 	if missing {
-		return fmt.Errorf("Found tests with no description")
+		return fmt.Errorf(errMissing)
+	}
+	if removed {
+		return fmt.Errorf(errRemoved)
 	}
 	return nil
 }
+
+func sliceContains(sl []string, s string) bool {
```

Comment:
This function does more than the name suggests: it also deletes the string if it is contained.
Can you name it accordingly?

Reply:
Also, what's the point of deletion?

Reply:
no need to iterate over tests that were already found in the json

Reply:
still, I don't get it. Why you remove from `allTests` if it's throw-away (and not exported outside)?